lib/soft_deletion/core.rb

```diff
@@ -94,12 +94,26 @@ def soft_delete_dependencies
 
     protected
 
+    def soft_delete_counter_cache_associations
+      Enumerator.new do |yielder|
+        each_counter_cached_associations do |association|
+          foreign_key = association.reflection.foreign_key.to_sym
+          unless destroyed_by_association && destroyed_by_association.foreign_key.to_sym == foreign_key
+            if send(association.reflection.name)
+              yielder << association
```

Comment:
or just yield here and don't build an enumerator ? .. then `soft_delete_counter_cache_associations(&: decrement_counters)`